### PR TITLE
Base architecture

### PR DESCRIPTION
**Pull request by Derovi**

### CODE REVIEW

**Comment by watislaf** [comment]
Add Read me with instructions
`gradle build`
`sudo docker-compose up -d`
`npm run build`
to build

File: docker-compose.yml [hunk 1]
```diff
@@ -36,6 +36,8 @@ services:
     depends_on:
       - auth
       - test
+    environment:
+      - REACT_APP_BASE_URL=http://api.localhost/
     ports:
```
Comment (watislaf) [inline docker-compose.yml]: is it used somewhere?

File: lindeck-frontend/README.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+### `npm run eject`
+
+**Note: this is a one-way operation. Once you `eject`, you can’t go back!**
+
+If you aren’t satisfied with the build tool and configuration choices, you can `eject` at any time. This command will remove the single build dependency from your project.
+
+Instead, it will copy all the configuration files and the transitive dependencies (webpack, Babel, ESLint, etc) right into your project so you have full control over them. All of the commands except `eject` will still work, but they will point to the copied scripts so you can tweak them. At this point you’re on your own.
+
+You don’t have to ever use `eject`. The curated feature set is suitable for small and middle deployments, and you shouldn’t feel obligated to use this feature. However we understand that this tool wouldn’t be useful if you couldn’t customize it when you are ready for it.
+
```
Comment (watislaf) [inline lindeck-frontend/README.md:41]: Remove this from readme. (Copy paste from npm documentation)

File: lindeck-frontend/README.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+Open [http://localhost:3000](http://localhost:3000) to view it in the browser.
+
+The page will reload if you make edits.\
+You will also see any lint errors in the console.
+
+### `npm test`
+
+Launches the test runner in the interactive watch mode.\
+See the section about [running tests](https://facebook.github.io/create-react-app/docs/running-tests) for more information.
+
```
Comment (watislaf) [inline lindeck-frontend/README.md:21]: are you going to write Tests?

File: lindeck-frontend/README.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+This section has moved here: [https://facebook.github.io/create-react-app/docs/making-a-progressive-web-app](https://facebook.github.io/create-react-app/docs/making-a-progressive-web-app)
+
+### Advanced Configuration
+
+This section has moved here: [https://facebook.github.io/create-react-app/docs/advanced-configuration](https://facebook.github.io/create-react-app/docs/advanced-configuration)
+
+### Deployment
+
+This section has moved here: [https://facebook.github.io/create-react-app/docs/deployment](https://facebook.github.io/create-react-app/docs/deployment)
+
```
Comment (watislaf) [inline lindeck-frontend/README.md:67]: is this necessary info to be in Read Me? (deletepls)

File: lindeck-frontend/src/index.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import './index.css';
+import App from './App';
+import axios from 'axios'
+
+axios.defaults.baseURL = "http://api.localhost/"
```
Comment (watislaf) [inline lindeck-frontend/src/index.js:7]: go set here our DNS name, so we don't need to change this line after each push to the Master branch

File: nginx.conf [hunk 6]
```diff
@@ -6,12 +6,12 @@ http {
     server {
         listen 80;
          server_name api.localhost;
```
Comment (watislaf) [inline nginx.conf:8]: DNS name too

File: lindeck-frontend/README.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+This project was bootstrapped with [Create React App](https://github.com/facebook/create-react-app).
+
+## Available Scripts
+
+In the project directory, you can run:
+
+### `npm start`
+
+Runs the app in the development mode.\
+Open [http://localhost:3000](http://localhost:3000) to view it in the browser.
```
Comment (watislaf) [inline lindeck-frontend/README.md:12]: Dns name

**Review by watislaf — CHANGES_REQUESTED** [review]
DNS name please (no localhost)
 Checked all web. Still waiting for the Discord discussion.

File: nginx.conf [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+            proxy_pass http://auth:8080/;
+         }
+         location /test/ {
+            proxy_pass http://test:8080/;
+         }
+    }
+
+    server {
+        listen 80;
+        server_name localhost;
```
Comment (watislaf) [inline nginx.conf:19]: www.localhost (www.domainname)

File: service-test/src/main/java/org/lindeck/lindeckserver/LindeckServerApplication.java [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+@ComponentScan(basePackages = "org.lindeck")
+@EnableJpaRepositories(basePackages="org.lindeck")
+@EntityScan(basePackages = "org.lindeck")
+public class LindeckServerApplication {
+
+	public static void main(String[] args) {
+		SpringApplication.run(LindeckServerApplication.class, args);
+	}
+
+}
```
Comment (watislaf) [inline service-test/src/main/java/org/lindeck/lindeckserver/LindeckServerApplication.java:19]: may be remove service-test and make service-api ? so after this push we can start using api

File: service-authorization/src/main/java/org/lindeck/authorization/controllers/MainController.java [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/")
+public class MainController {
+    @GetMapping(produces = MediaType.APPLICATION_JSON_VALUE)
+    String test() {
+        return "test";
+    }
+}
```
Comment (watislaf) [inline service-authorization/src/main/java/org/lindeck/authorization/controllers/MainController.java:15]: image
where is Test :<

File: lindeck-frontend/build.sh [hunk 11]
```diff
@@ -0,0 +1,4 @@
+react-scripts build
+sudo rm -f -r /var/www/html/*
+sudo mkdir -p /var/www/html
+sudo mv build/* /var/www/html
```
Comment (watislaf) [inline lindeck-frontend/build.sh:4]: :? for what

**Comment by watislaf** [comment]
delete package-json for 17000 lines

File: docker-compose.yml [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+  test:
+    build:
+      context: service-test/
+    ports:
+      - 8071:8080
+    links:
+      - psql
+    depends_on:
+      - psql
+
```
[the remark this answers is not in the document]
Comment (Derovi) [reply docker-compose.yml:30]: It's like test

File: gradle/wrapper/gradle-wrapper.properties [hunk 13]
```diff
@@ -0,0 +1,5 @@
+distributionBase=GRADLE_USER_HOME
+distributionPath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.5-bin.zip
+zipStoreBase=GRADLE_USER_HOME
+zipStorePath=wrapper/dists
```
[the remark this answers is not in the document]
Comment (Derovi) [reply gradle/wrapper/gradle-wrapper.properties:5]: Automatic

File: module-data/src/main/java/org/lindeck/data/dao/UserDAO.java [hunk 14]
```diff
@@ -0,0 +1,12 @@
...
+
+import org.lindeck.data.model.User;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface UserDAO extends CrudRepository<User, String> {
+    boolean existsByLogin(String login);
+    boolean existsByEmail(String email);
+    User findByLogin(String login);
```
[the remark this answers is not in the document]
Comment (Derovi) [reply module-data/src/main/java/org/lindeck/data/dao/UserDAO.java:11]: For checking if user exists by login / email

File: docker-compose.yml (hunk 1 above)
Comment (Derovi) [reply docker-compose.yml]: No!

File: lindeck-frontend/README.md (hunk 3 above)
Comment (Derovi) [reply lindeck-frontend/README.md:21]: No!

File: docker-compose.yml (hunk 1 above)
Comment (watislaf) [reply docker-compose.yml]: So remove it

File: service-test/src/main/java/org/lindeck/lindeckserver/LindeckServerApplication.java (hunk 9 above)
Comment (Derovi) [reply service-test/src/main/java/org/lindeck/lindeckserver/LindeckServerApplication.java:19]: There are could be many services in API (microservices), so test is just an example of such service.

File: service-authorization/src/main/java/org/lindeck/authorization/controllers/MainController.java (hunk 10 above)
Comment (Derovi) [reply service-authorization/src/main/java/org/lindeck/authorization/controllers/MainController.java:15]: Did you try to sign-in?

File: lindeck-frontend/build.sh (hunk 11 above)
Comment (Derovi) [reply lindeck-frontend/build.sh:4]: Auto-generated

File: service-authorization/src/main/java/org/lindeck/authorization/controllers/MainController.java (hunk 10 above)
Comment (watislaf) [reply service-authorization/src/main/java/org/lindeck/authorization/controllers/MainController.java:15]: go push the last updates I will test

File: lindeck-frontend/README.md (hunk 3 above)
Comment (watislaf) [reply lindeck-frontend/README.md:21]: So remmove ?

File: lindeck-frontend/src/index.js (hunk 5 above)
Comment (watislaf) [reply lindeck-frontend/src/index.js:7]: Забей на днс, когда дэдик сделам поставим

File: nginx.conf (hunk 8 above)
Comment (watislaf) [reply nginx.conf:19]: www !!

File: docker-compose.yml (hunk 1 above)
Comment (Derovi) [reply docker-compose.yml]: I Will!

File: lindeck-frontend/src/index.js (hunk 5 above)
Comment (Derovi) [reply lindeck-frontend/src/index.js:7]: Please, use English. I can't use translate for every conversation =(

File: service-authorization/src/main/java/org/lindeck/authorization/controllers/MainController.java (hunk 10 above)
Comment (Derovi) [reply service-authorization/src/main/java/org/lindeck/authorization/controllers/MainController.java:15]: Try to sign-in!

**Review by watislaf — APPROVED** [review]
well, you can login several times, he does not check for this
it is not clear where the damn jwt token is and how to use it
 
but for Christ's sake, I hope it will get better in the next request

**Pull request merged**